client/data/hosting/use-cache.ts

```diff
@@ -48,14 +51,11 @@ export const purgeEdgeCache = async ( siteId: number ) => {
 	} );
 };
 
-export const useEdgeCacheQuery = ( siteId: number ) => {
-	return useQuery< boolean, unknown, boolean >( {
+export const useEdgeCacheQuery = ( siteId: number | null ) => {
+	return useQuery< boolean >( {
 		queryKey: [ USE_EDGE_CACHE_QUERY_KEY, siteId ],
-		queryFn: () => getEdgeCacheStatus( siteId ),
+		queryFn: () => getEdgeCacheStatus( siteId as number ),
```

Comment:
Not super happy with this `as` declaration, but the `enabled: !! siteId` directive should make it safe, at least.